include/fuzzy/suffix_array_index.hh

```diff
@@ -15,7 +15,11 @@ namespace fuzzy
   class SuffixArrayIndex
   {
   public:
+#ifdef _WIN32
```

Comment:
We should try to avoid a `ifdef` for this simple case.

Can you move the constant out of the class definition and mark it as `constexpr`?

```cpp
namespace fuzzy
{
  constexpr size_t DEFAULT_MAX_TOKENS_IN_PATTERN = 300;

  class SuffixArrayIndex
  ...
}
```

Reply:
Thanks for reviewing, I just rebased code, Please check and merge code.